packages/react/src/components/Alert/Alert.tsx

```diff
@@ -0,0 +1,138 @@
+  closeButton?: ShorthandValue<ButtonProps>
+
+  /** An alert may be formatted to display a danger message. */
+  danger?: boolean
+
+  /** An alert may be formatted to display information. */
+  info?: boolean
+
+  /**
+   * The callback used when the user chooses to hide the Alert.
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props and proposed value.
+   */
+  onClose?: ComponentEventHandler<AlertProps>
+
+  /** An alert may be formatted to display a successful message. */
+  success?: boolean
+
+  /** An alert may be formatted to display a warning message. */
+  warning?: boolean
+}
+
+/**
+ * A Alert displays information that explains nearby content.
+ */
+class Alert extends UIComponent<ReactProps<AlertProps>> {
+  static displayName = 'Alert'
+  static className = 'ui-alert'
+
+  static slotClassNames: AlertSlotClassNames = {
+    content: `${Alert.className}__content`,
+    closeButton: `${Alert.className}__closeButton`,
+  }
+
+  static propTypes = {
+    ...commonPropTypes.createCommon(),
+    attached: PropTypes.oneOfType([PropTypes.bool, PropTypes.oneOf(['top', 'bottom'])]),
+    closeButton: customPropTypes.itemShorthand,
+    danger: PropTypes.bool,
+    info: PropTypes.bool,
+    onClose: PropTypes.func,
+    success: PropTypes.bool,
+    warning: PropTypes.bool,
+  }
+
+  static defaultProps = {
+    accessibility: defaultBehavior,
+    closeButton: { icon: 'close' },
+  }
+
+  renderComponent(config: RenderResultConfig<AlertProps>) {
+    const { accessibility, classes, ElementType, unhandledProps } = config
+    const { children, content } = this.props
+    return (
+      <ElementType
+        className={classes.root}
+        {...accessibility.attributes.root}
+        {...rtlTextContainer.getAttributes({ forElements: [children, content] })}
+        {...unhandledProps}
+      >
+        {childrenExist(children) ? children : this.renderContent(config)}
+      </ElementType>
+    )
+  }
+
+  renderContent = ({ styles }: RenderResultConfig<AlertProps>) => {
+    const { closeButton, content, onClose } = this.props
+    return (
+      <>
+        {Box.create(content, {
+          defaultProps: {
+            className: Alert.slotClassNames.content,
+            styles: styles.content,
+          },
+        })}
+        {onClose &&
+          Button.create(closeButton, {
+            defaultProps: {
+              iconOnly: true,
+              text: true,
+              className: Alert.slotClassNames.closeButton,
+              styles: styles.closeButton,
+            },
+            overrideProps: this.handleCloseButtonOverrides,
+          })}
+      </>
+    )
+  }
+
+  handleCloseButtonOverrides = (predefinedProps: ButtonProps) => ({
+    onClick: (e: React.SyntheticEvent, buttonProps: ButtonProps) => {
+      this.handleOnClose(e)
+      _.invoke(predefinedProps, 'onClick', e, buttonProps)
+    },
+  })
+
+  handleOnClose = (e: React.SyntheticEvent) => {
```

Comment:
Can you please also:
- place `handleCloseButtonOverrides` and `handleOnClose` before `renderComponent`
- place `renderContent` before `renderComponent`

---

- `handleCloseButtonOverrides`
- `handleOnClose`
- `renderContent`
- `renderComponent`

Reply:
done